src/CMakeLists.txt

```diff
@@ -35,26 +38,26 @@ set(NIMBLE_DARMA_SOURCES
 )
 
 add_executable(NimbleSM_Serial nimble_serial.cc ${NIMBLE_SOURCES} ${NIMBLE_EXTRAS_SOURCES})
-target_link_libraries(NimbleSM_Serial ${NIMBLE_KOKKOS_LIBS} ${NIMBLE_EXTRAS_LIBS} ${Trilinos_TPL_LIBRARIES} ${Trilinos_LIBRARIES} ${Exodus_LIBS} ${bvh_LIBS})
+target_link_libraries(NimbleSM_Serial ${MPI_CXX_LIBRARIES} ${NIMBLE_KOKKOS_LIBS} ${NIMBLE_EXTRAS_LIBS} ${Trilinos_TPL_LIBRARIES} ${Trilinos_LIBRARIES} ${Exodus_LIBS} ${bvh_LIBS})
```

Comment:
I don't think it will hurt but does NimbleSM_Serial need to link to MPI? It shouldn't have to unless there is a problem somewhere in the code.

Reply:
The issue is that the we compile all object files either with or without MPI, thanks to the compiler definitions. So some source files that are included in the Nimble build will have MPI dependencies...perhaps there is an issue with the source files that are included in the serial executable build? But otherwise, we get undefined reference errors on link for MPI routines without this change.